src/ucp/tag/rndv.c

```diff
@@ -469,15 +469,21 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_progress_rma_get_zcopy, (self),
         length = ucs_min(chunk, rndv_req->send.length - offset);
     }
 
+    /* ensure that the current length is over min_zcopy */
+    length = ucs_max(length, min_zcopy);
```

Comment:
can you describe the failing case plz?
Should not rndv fallback to PUT scheme if the data length is less than `min_zcopy`?

Reply:
> can you describe the failing case plz?

> Should not rndv fallback to PUT scheme if the data length is less than min_zcopy?

yes, this is done for the sizes < minimum GET Zcopy (i.e. `[0..64]`)
but for `[65..inf)`, the chunk can be lower than minimum GET Zcopy
For example, it was reproduced for the message with the size - 128 bytes. it was split to two chunks (`32`, `96`). We take care of the last chunk, but we should also adjust the current one.

Reply:
maybe better do not split such small packets at all?

Reply:
> maybe better do not split such small packets at all?

yes, it can be better
but it may happen if we have two devices with the BW:
- 1 MB/s
- 200 GB/s
the RNDV scale will be:
- 1 / 204800 = 0.000005
- 1

so, a message with size = 1 MB will have the following chunks:
- 1MB / 2 * `RNDV scale for 0 lane` = 512KB * 0.000005 = 3 bytes // so, this is lower than minimum GET Zcopy
- (1MB - 3B) / 2 * `RNDV scale for 1 lane` = 512KB * 1 = ~512 bytes
- ...

Also, for instance, minimum GET Zcopy could be 1 KB for some new transport that will be added to UCT.
BTW, probably, we should have some threshold from which we will start RNDV multi-rail if a user sets `UCX_RNDV_THRESH=0`.

Reply:
let's concentrate on the existing ones for now. 
i think we need to introduce some minimum thresh for using multirail instead of fixing splitting small messages